src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,434 @@
+                        tokens[0].Equals(CodeCoverageString, StringComparison.OrdinalIgnoreCase))
+                    {
+                        isCollectCodeCoverageEnabled = true;
+                    }
+
+                    allArgs.Add("--collect:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
+                }
+            }
+
+            if (isCollectCodeCoverageEnabled || isRunSettingsEnabled)
+            {
+                // Pass TraceDataCollector path to vstest.console as TestAdapterPath if --collect "Code Coverage"
+                // or --settings (User can enable code coverage from runsettings) option given.
+                // Not parsing the runsettings for two reason:
+                //    1. To keep no knowledge of runsettings structure in VSTestTask.
+                //    2. Impact of adding adapter path always is minimal. (worst case: loads additional data collector assembly in datacollector process.)
+                // This is required due to currently trace datacollector not ships with dotnet sdk, can be remove once we have
+                // go code coverage x-plat.
+                if (!string.IsNullOrEmpty(VSTestTraceDataCollectorDirectoryPath))
+                {
+                    allArgs.Add("--testAdapterPath:" +
+                                ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(
+                                    VSTestTraceDataCollectorDirectoryPath));
+                }
+            }
+
+            if (!string.IsNullOrEmpty(VSTestNoLogo))
+            {
+                allArgs.Add("--nologo");
+            }
+
+            if (!string.IsNullOrEmpty(VSTestArtifactsProcessingMode) && VSTestArtifactsProcessingMode.Equals("collect", StringComparison.OrdinalIgnoreCase))
+            {
+                allArgs.Add("--artifactsProcessingMode-collect");
+            }
+
+            if (!string.IsNullOrEmpty(VSTestSessionCorrelationId))
+            {
+                allArgs.Add("--testSessionCorrelationId:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(VSTestSessionCorrelationId));
+            }
+
+            return allArgs;
+        }
+
+        private int ExecuteTest()
+        {
+            string packagePath = $@"{NugetPath}\microsoft.testplatform\{VSTestRunnerVersion}\tools\net462\Common7\IDE\Extensions\TestPlatform\";
+
+            var processInfo = new ProcessStartInfo
+            {
+                FileName = $"{packagePath}vstest.console.exe",
+                Arguments = string.Join(" ", CreateArguments()),
+                UseShellExecute = false,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+            };
+
+            using var activeProcess = new Process { StartInfo = processInfo };
+            activeProcess.Start();
+            using StreamReader errReader = activeProcess.StandardError;
```

Comment:
This deadlocks for me.  Moving to ToolTask should fix things

Reply:
Updated to be a tooltask